### PR TITLE
Getting default values from data in formwidget

### PR DESCRIPTION
**Pull request by HammenWS**
When adding a new repeater field in update context, default values for the added fields won't be respected.

Some issues where this was mentioned:
https://github.com/octobercms/october/issues/2445#issuecomment-274621932
#2013 

You can replicate this in the test plugin by adding a default value for some field in the config/repeater_fields.yaml 

When you *create* a new item using this repeater field, the default value will be set.
But when you *update* an item, the default value won't be set when adding a new group.

This PR should fix this problem.

### CODE REVIEW

File: modules/backend/widgets/Form.php [hunk 1]
```diff
@@ -1069,13 +1069,21 @@ protected function getFieldValue($field)
...
 
         return $field->getValueFromData($this->data, $defaultValue);
     }
 
+    /**
+     * Checks if default values should be taken from data.
+     * This should be done when model exists or when explicitly configured
+     */
+    protected function shouldFetchDefaultValues() {
+        return !$this->model->exists || object_get($this->config, 'shouldFetchDefaultValues');
```
Comment (LukeTowers) [inline modules/backend/widgets/Form.php]: If we're using a config value for this it should support being explicitly set to false and having that override `!$this->model->exists`

**Comment by HammenWS** [comment]
@LukeTowers 
I didn't think about such a usecase, but I understand why this is wanted. I added the changes to this PR.

**Comment by HammenWS** [comment]
@LukeTowers Hello? Have you seen my changes?

**Comment by LukeTowers** [comment]
@HammenWS could you post the YAML file where this option would be used?

**Comment by LukeTowers** [comment]
Could you rename the `shouldFetchDefaultValues` config property (and related variables) to: `enableDefaults`? And then add documentation (and necessary supporting code to the FormBehavior) to https://octobercms.com/docs/backend/forms#configuring-form for how people can use `enableDefaults` (not defined, based on whether the model exists or not - default behavior; true, always use them; false, never use them)

**Comment by LukeTowers** [comment]
Closing in favour of #4098

**Pull request closed without merging**